pkg/platform/openstack/kuryr_bootstrap.go

```diff
@@ -283,6 +286,21 @@ func getWorkersSubnetFromMasters(client *gophercloud.ServiceClient, kubeClient c
 	return findOpenStackSubnetByNetworkTag(client, primaryNetworkTag)
 }
 
+// Check if the Cloud auth endpoint is set to not use the Proxy
+func shouldCloudUseProxy(no_proxy, auth string) (bool, error) {
+	auth_url, err := url.Parse(auth)
+	if err != nil {
+		return false, err
+	}
+	no_proxy_addresses := strings.Split(no_proxy, ",")
+	for _, address := range no_proxy_addresses {
+		if auth_url.Hostname() == address {
+			return false, nil
+		}
+	}
```

Comment:
Uhm, the logic might need to be more complicated. Something like `.openshift.org` should probably match `foo.openshift.org`.

Reply:
true, thanks